src/chai/ManagedArray.hpp

```diff
@@ -454,21 +454,36 @@ ManagedArray<T> makeManagedArray(T* data,
                                  ExecutionSpace space,
                                  bool owned)
 {
+#if !defined(CHAI_DISABLE_RM)
   ArrayManager* manager = ArrayManager::getInstance();
 
-  PointerRecord* record =
-      manager->makeManaged(data, sizeof(T) * elems, space, owned);
-
-  for (int space = CPU; space < NUM_EXECUTION_SPACES; space++) {
-    record->m_allocators[space] = 
-      manager->getAllocatorId(ExecutionSpace(space));
+  // First, try and find an existing PointerRecord for the pointer
+  PointerRecord* record = manager->getPointerRecord(data);
+  bool existingRecord = true;
+  if (record == &ArrayManager::s_null_record) {
+    // create a new pointer record for external pointer
+    record = manager->makeManaged(data, sizeof(T) * elems, space, owned);
+    existingRecord = false;
   }
+  ManagedArray<T> array(record, space);
 
-  ManagedArray<T> array = ManagedArray<T>(record, space);
+  if (existingRecord && !owned) {
+    // pointer has an owning PointerRecord, create a non-owning ManagedArray
+    // slice
+    array = array.slice(0, elems);
+  }
 
   if (!std::is_const<T>::value) {
     array.registerTouch(space);
   }
+#else
+  PointerRecord recordTmp;
```

Comment:
I just realized, I don't think we use pointer records when CHAI_DISABLE_RM is defined. They are usually set to nullptr.

Reply:
The PointerRecord gets used in the CHAI_DISABLE_RM codepath just to pass the pointers and slice/ownership status to the ManagedArray constructor; the underlying m_pointer_record field in the ManagedArray is still set to nullptr.